src/features/transactions/transactionsSlice.ts

```diff
@@ -51,16 +65,28 @@ const initialState: TransactionsState = {
 
 function updateTransaction(
   state: TransactionsState,
+  nonce: string,
   hash: string,
-  status: "processing" | "succeeded" | "reverted"
+  signerWallet: string,
+  status: "processing" | "succeeded" | "reverted",
+  protocol?: "request-for-quote" | "last-look"
```

Comment:
I think we should add types or enums for status and protocol types.

Reply:
Agree, got a little tedious typing that in.